### PR TITLE
Prod deploy pipeline

### PR DESCRIPTION
**Pull request by ghost**
### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/832

### What changes does this PR make to Grapl? Why?
This adds a deploy pipeline for production, represented by the mock customer accounts.

This includes the following changes:
Adding the mock customer pulumi config files
Having the merge pipeline automatically add artifacts to the mock customer config files just like it does for staging
Creating an actual production pipeline to deploy to these accounts

### How were these changes tested?
Manual deploys

### CODE REVIEW

File: .buildkite/pipeline.deploy.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+steps:
+  - label: ":pulumi: Update grapl/customer-mock1 environment in customer-mock1 account"
+    plugins:
+      - grapl-security/vault-login#v0.1.0
+      - grapl-security/vault-env#v0.1.0:
+          secrets:
+            - PULUMI_ACCESS_TOKEN
+      - improbable-eng/metahook#v0.4.1:
+          post-command: |
+            .buildkite/scripts/grapl_testing_stack/dump_artifacts.sh
```
Comment (christophermaier) [inline .buildkite/pipeline.deploy.yml:15]: I don't think this plugin is necessary here.

File: .buildkite/pipeline.deploy.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - grapl-security/pulumi#v0.1.2:
+          command: update
+          project_dir: pulumi/grapl
+          stack: grapl/customer-mock1
+    agents:
+      queue: "pulumi-customer-mock1"
+    artifact_paths:
+      - "test_artifacts/**/*"
```
Comment (christophermaier) [inline .buildkite/pipeline.deploy.yml:23]: This `artifact_paths` stanza isn't needed either.

File: .buildkite/pipeline.deploy.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - grapl-security/vault-env#v0.1.0:
+          secrets:
+            - PULUMI_ACCESS_TOKEN
+      - improbable-eng/metahook#v0.4.1:
+          post-command: |
+            .buildkite/scripts/grapl_testing_stack/dump_artifacts.sh
+      - grapl-security/pulumi#v0.1.2:
+          command: update
+          project_dir: pulumi/grapl
+          stack: grapl/customer-mock1
```
Comment (christophermaier) [inline .buildkite/pipeline.deploy.yml:37]: This should be `grapl/customer-mock2`

**Review by christophermaier — APPROVED** [review]
Looks good, pending some minor tweaks.

I appreciate the breakdown of the commits, too 👍

File: .buildkite/pipeline.deploy.yml (hunk 1 above)
Comment (twunderlich-grapl) [reply .buildkite/pipeline.deploy.yml:15]: If it turns out that we need these logs, this is easy enough to add back in. And frankly they're probably available at a different artifact path anyway. Removing

Comment (christophermaier) [reply .buildkite/pipeline.deploy.yml]: These logs are generated from our integration / e2e tests, though, not Pulumi.

**Pull request merged**